### ISSUE TITLE
MethodInfo.typeDescriptorStr is useful but is private

### ISSUE THREAD

**Issue by rankinc** [classgraph/classgraph#380]
```java
MethodInfo {

    /**
     * The JVM-internal type descriptor (missing type parameters, but including types for synthetic and mandated
     * method parameters).
     */
    private String typeDescriptorStr;

}
```
This field is useful precisely _because_ it is read directly from the. Java classfile without being filtered through ClassGraph's "presentation layer" i.e. `MethodTypeSignature.toString()`, whose output cannot be relied upon to remain constant as ClassGraph evolves.

A supported API for retrieving the internal type descriptor string would spare me from finding ways to extract it via reflection.

And ditto for `Fieldinfo.typeDescriptorStr`, most likely.

**Background**
We need to be able to configure our tool to be able to ignore certain class methods, some of which are synthetic Kotlin methods which cannot be annotated. Therefore the only _reliable_ way of identifying these methods is by their `name` + `internal descriptor`.

**Comment by lukehutch** [comment]
@rankinc OK, I exposed both type descriptor and type signature strings in `FieldInfo` and `MethodInfo`, and type signature strings in `ClassInfo` (there are no type descriptor strings for classes).

Let me know if you can think of anything else you need before I push out a new version.

**Comment by rankinc** [comment]
Thanks, that works :smile:

**Comment by lukehutch** [comment]
Thanks, will be released in 4.8.53 once the other issue is resolved.

**Comment by lukehutch** [comment]
Released in 4.8.53. Thanks for the report!
